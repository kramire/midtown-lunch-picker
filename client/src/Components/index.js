import Reviews from './Reviews';
import LocationDetails from './LocationDetails';
import RandomGenerator from './RandomGenerator';
import ReviewItem from './ReviewItem';
import BackgroundImage from './BackgroundImage';
import StarRating from './StarRating';

export {
  Reviews,
  LocationDetails,
  RandomGenerator,
  ReviewItem,
  BackgroundImage,
  StarRating,
};
