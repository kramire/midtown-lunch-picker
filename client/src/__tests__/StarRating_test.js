import React from 'react';
import { shallow, mount, unmount } from 'enzyme';
import renderer from 'react-test-renderer';
import { StarRating } from '../Components';

describe('StarRating Component Testing', () => {
  it('Smoke Test --> should render without crashing, with default rating of 0', () => {
    // Arrange

    // Act
    const component = shallow(<StarRating />);

    // Assert
    expect(component.find(`[test-id="star-rating"]`)).toHaveLength(1);
    expect(component.find(`[test-id="star-icon"]`)).toHaveLength(0);
  });

  it('Shallow Render Test--> 4.5 rating --> should render 4 children', () => {
    // Arrange
    const rating = 4.5;

    // Act
    const component = shallow(<StarRating rating={rating} />);

    // Assert
    expect(component.find(`[test-id="star-icon"]`)).toHaveLength(4);
  });

  it('Full Render Test --> 4.5 rating --> should render 4 children', () => {
    // Arrange
    const rating = 4.5;

    // Act
    const component = mount(<StarRating rating={rating} />);

    // Assert
    expect(component.find(`[test-id="star-icon"]`)).toHaveLength(4);
    component.unmount();
  });

  it('Snapshot Test --> no rating --> should render no children', () => {
    // Arrange

    // Act
    const component = renderer
      .create(<StarRating />)
      .toJSON();

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('Snapshot Test --> 4.5 rating --> should render 4 children', () => {
    // Arrange
    const rating = 4.5;

    // Act
    const component = renderer
      .create(<StarRating rating={rating} />)
      .toJSON();

    // Assert
    expect(component).toMatchSnapshot();
  });
});
