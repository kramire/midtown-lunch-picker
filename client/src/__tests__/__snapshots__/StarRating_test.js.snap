// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating Component Testing Snapshot Test --> 4.5 rating --> should render 4 children 1`] = `
<div
  test-id="star-rating"
>
  <span
    className="fa fa-star"
    test-id="star-icon"
  />
  <span
    className="fa fa-star"
    test-id="star-icon"
  />
  <span
    className="fa fa-star"
    test-id="star-icon"
  />
  <span
    className="fa fa-star"
    test-id="star-icon"
  />
</div>
`;

exports[`StarRating Component Testing Snapshot Test --> no rating --> should render no children 1`] = `
<div
  test-id="star-rating"
/>
`;
